TrackletAlgorithm/globalFunctions.h

```diff
@@ -0,0 +1,16 @@
+#ifndef TrackletAlgorithm_globalFunctions_h
+#define TrackletAlgorithm_globalFunctions_h
+
+#ifdef CMSSW_GIT_HASH
+
+ static std::string decodeToBits(unsigned int field, unsigned int size) {
```

Comment:
@carriganm95 Is there any point in this being `static`, now that it is outside of a class?

Also, have you tried making this a template function, as I suggested?
```c++
template<class T>
static std::string decodeToBits(const T field) {
    constexpr unsigned int size = T::width;
    …
```
I think this would make the function more foolproof, since you don't need to give the size separately. And since the template parameter is the type of an argument, you shouldn't need to give it explicitly when calling the function, i.e., you can just call it as
```c++
decodeToBits(getR());
```
for example.

Reply:
I have implemented this templated function.